### ISSUE TITLE
Ask users permission for additional communications

### ISSUE THREAD

**Issue by kazlauskis** [NERC-CEH/irecord-butterflies-app#187]
Add an opt in tick box and message for additional (marketing) communications from Butterfly Conservation.

**Comment by kazlauskis** [comment]
For reference. I have added an extra iRecord user profile field, `field_agree_bc_coms`. This is an optional question that will appear on the user profile page but won't be asked during the registration on the website.

My next steps will be to incorporate this into the iRecord Butterflies app, requiring every user to specify their choice before using it.

![Image](https://github.com/user-attachments/assets/66d903e9-2f19-4ca7-b0ea-cf37b19ecf9a)

**Comment by kazlauskis** [comment]
@SmithButterfly @RichardFoxBC I am not entirely sure how you will use this permission. Could you please provide a brief description (1-3 sentences) explaining to the user what this is about? I'll then add an app popup and the option on the app settings page.

**Comment by SmithButterfly** [comment]
Hi @kazlauskis yes, I have some wording that I'll get agreed and post to you shortly, thanks

**Comment by SmithButterfly** [comment]
Hi @kazlauskis here is the wording -  just need to confirm the privacy notice to link to. I will update. We had a few questions too before it would go live please - I'll post those separately.

Please tick if you would like to receive Butterfly Conservation communications [unchecked tick box]
Butterfly Conservation would love to keep you updated with news about campaigns and other work by email. You can change your preferences at any time. Butterfly Conservation will keep your details safe and will never swap, sell or rent your details to anyone. For more details about this, see Butterfly Conservation’s Privacy Policy [linked]. Please note this opt in does not include the communications you may receive directly about your records as detailed in the iRecord.org.uk privacy notice.

**Comment by SmithButterfly** [comment]
@kazlauskis Will the push notification with the new communications consent only go out to an app user when a they download the update, or will it will go to all users regardless of app version? 
For planning, capacity to manage the initial mail-list and get out the first interesting content will be limited during March. Can we be flexible on the date of this going out or will it be pushed out as soon as the updates are finalised? thanks

**Comment by SmithButterfly** [comment]
@kazlauskis does the consent opt-in remain as a changeable option on the account settings after registration (or push notification)? Considering the management of opt outs and updates. Users can opt out via each email they receive.

**Comment by kazlauskis** [comment]
> Will the push notification with the new communications consent only go out to an app user when a they download the update, or will it will go to all users regardless of app version?

@SmithButterfly The request popup will only appear for users with the new app version upon their first opening.

> For planning, capacity to manage the initial mail-list and get out the first interesting content will be limited during March. Can we be flexible on the date this will go out, or will it be pushed out as soon as the updates are finalised? thanks

I can set a timer to ensure the popup only appears after a specific date, if that helps.

> does the consent opt-in remain as a changeable option on the account settings after registration (or push notification)? Considering the management of opt outs and updates. Users can opt out via each email they receive.

Users will be able to change their choice on the app's settings page as well as on their user profile page on the iRecord website. I think we should include an email opt-out (unsubscribe) link in the emails, but I don't know how this can be implemented to automatically change the user's settings on the iRecord site.

**Comment by DavidRoy** [comment]
@kazlauskis we've had the go ahead on this from the ukceh data officer.  
How we identify iRecord butterfly app users who have reviewed the new option to opt in/out of BC communications.  Will the users be required to confirm the new privacy term and you will then populate the BC comms tick box with a true or false.  Until users review the new terms and for other iRecord users, the BC comms check-box will remain ‘null’.  If you can confirm this, we can setup a database query to extract the necessary information to pass onto BC – i.e. a list of users who have reviewed the new app privacy notice

**Comment by kazlauskis** [comment]
@DavidRoy, that's right, the `field_agree_bc_coms` field is empty by default, and once set, it will be `true` or `false`.  Users will be presented a popup and then I will set the value accordingly. It is worth noting that users can also change the value on the website, and there they might not see the link to the BC terms.

Is there a reason why you want to identify ALL the users who have reviewed the option? Only users with `field_agree_bc_coms=true` in their profiles should receive BC emails.

Also, GDPR might require you to provide an `unsubscribe` button in the comms emails. I don't know if the unsubscribing has to be a fully automated process or simply linking to the user profile where they can update the setting is enough.

**Comment by DavidRoy** [comment]
@kazlauskis - you are right that we might only want to report field_agree_bc_coms = true
@SmithButterfly - do you need to see a list of field_agree_bc_coms = false as well?

**Comment by SmithButterfly** [comment]
@DavidRoy and @kazlauskis that's right we'd only need the field_agree_bc_coms = true and then this list is compared with all lists in the mailing system. Each emailed newsletter, whether to this list or another BC list, includes an opt out option. So if anyone still has the opt in flagged as true but has since opted out via the email, they will not receive another email. As users can change their preference in the settings at any time we'll need to download the whole opt in list via the database query each time, not just changes.

@kazlauskis the delayed timing on the pop up would be helpful. I will check the best date

**Comment by kazlauskis** [comment]
@SmithButterfly I added the message in the app. It is quite long, but that might be acceptable. Are you OK with this? Do you have a link to the privacy policy? Also, from what date would you like me to display this?

I have changed the user profile field name to `field_agree_bc_comms`.

![Image](https://github.com/user-attachments/assets/4fbcfc52-a5c8-41e7-a271-0a7e45e0e61a)

**Comment by kazlauskis** [comment]
@DavidRoy @SmithButterfly, I'd like to release the app for Beta testing. Are you happy with how I've implemented this?

**Comment by SmithButterfly** [comment]
Hi @kazlauskis and @DavidRoy , thank you, it looks great. I don't have a tailored privacy notice, but this one will work and could be updated in future with a streamlined version (as this includes cookie notice relevant to the Butterfly Conservation website only)
https://butterfly-conservation.org/privacy-and-cookies
